octopusdeploy/schema_kubernetes_agent_worker.go

```diff
@@ -0,0 +1,157 @@
+package octopusdeploy
+
+import (
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/machines"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"net/url"
+)
+
+func expandKubernetesAgentWorker(kubernetesAgent *schema.ResourceData) *machines.Worker {
+	uri, _ := url.Parse(kubernetesAgent.Get("uri").(string))
+	thumbprint := kubernetesAgent.Get("thumbprint").(string)
+
+	communicationsMode := kubernetesAgent.Get("communication_mode").(string)
+	upgradeLocked := kubernetesAgent.Get("upgrade_locked").(bool)
+	var endpoint machines.IEndpoint = machines.NewKubernetesTentacleEndpoint(uri, thumbprint, upgradeLocked, communicationsMode, "")
```

Comment:
I know very little about the Kubernetes agent, should the namespace be set here?

Reply:
The `defaultNamespace` is only applicable to the Kubernetes agent when it is operating as a deployment target. It defines the Kubernetes namespace to which applications get deployed.
In this case the Kubernetes agent is operating as a worker which means that it does not have this concept. The namespace that the worker itself resides in is defined separately using the [namespace of the Helm installation](https://github.com/OctopusDeploy/helm-charts/blob/main/charts/kubernetes-agent/templates/tentacle-deployment.yaml#L5).